pkg/metrics/metrics_test.go

```diff
@@ -41,41 +42,69 @@ func CreateDiskWithConfidentialCompute(betaVersion bool, confidentialCompute boo
 	})
 }
 
-func TestGetEnableConfidentialCompute(t *testing.T) {
+func CreateDiskWithStoragePool(storagePool string, diskType string) *gce.CloudDisk {
+	return gce.CloudDiskFromAlpha(&computealpha.Disk{
+		StoragePool: storagePool,
+		Type:        diskType,
+	})
+}
+
+func TestGetMetricParameters(t *testing.T) {
 	testCases := []struct {
 		name                              string
 		disk                              *gce.CloudDisk
 		expectedEnableConfidentialCompute string
 		expectedDiskType                  string
+		expectedEnableStoragePools        string
 	}{
 		{
 			name:                              "test betaDisk with enableConfidentialCompute=false",
 			disk:                              CreateDiskWithConfidentialCompute(true, false, hyperdiskBalanced),
 			expectedEnableConfidentialCompute: "false",
 			expectedDiskType:                  hyperdiskBalanced,
+			expectedEnableStoragePools:        "false",
 		},
 		{
 			name:                              "test betaDisk with enableConfidentialCompute=true",
 			disk:                              CreateDiskWithConfidentialCompute(true, true, hyperdiskBalanced),
 			expectedEnableConfidentialCompute: "true",
 			expectedDiskType:                  hyperdiskBalanced,
+			expectedEnableStoragePools:        "false",
 		},
 		{
-			name:                              "test disk withpit enableConfidentialCompute",
+			name:                              "test disk without enableConfidentialCompute",
 			disk:                              CreateDiskWithConfidentialCompute(false, false, hyperdiskBalanced),
 			expectedEnableConfidentialCompute: "false",
 			expectedDiskType:                  hyperdiskBalanced,
+			expectedEnableStoragePools:        "false",
+		},
+		{
+			name:                              "test alphaDisk with storage pool projects/my-project/zone/us-central1-a/storagePools/sp1",
+			disk:                              CreateDiskWithStoragePool("projects/my-project/zone/us-central1-a/storagePools/sp1", hyperdiskBalanced),
+			expectedEnableConfidentialCompute: "false",
+			expectedDiskType:                  hyperdiskBalanced,
+			expectedEnableStoragePools:        "true",
+		},
+		{
+			name:                              "test alphaDisk with no storage pool",
+			disk:                              CreateDiskWithStoragePool("", hyperdiskBalanced),
```

Comment:
Can we add a test case that if the GetDisk call return an error(disk object will be empty), what is the expected output of this metrics?

Reply:
Good call. Added.